src/pages/Settings/Settings.js

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react'
+import useForm from 'react-hook-form'
+import * as yup from 'yup'
+import Button from 'components/Button'
+
+import './Settings.module.css'
+
+export const PORT_NUMBER_REGEX = /^\d{4,6}$/
+export const EMAIL_REGEX = /^\S+@\S+$/i
+
+const SettingsValidationSchema = yup.object().shape({
+  hostName: yup.string().required(),
+  hostPubKey: yup.string().required(),
+  registrationEmail: yup.string()
+    .email()
+    .required(),
+  deviceName: yup.string().required(),
+  networkId: yup.string().required(),
+  deviceAdminPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hcAdminPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hcNetworkPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hostingPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required()
+})
+
+export function Settings ({
+  settings,
+  updateSettings,
+  factoryReset,
+  toggleSshAccess,
+  history: { push }
+}) {
+  const goToMenu = () => push('/menu')
+  const { register, handleSubmit, errors } = useForm({
+    defaultValues: settings,
+    validationSchema: SettingsValidationSchema
+  })
+
+  const [sshAccessVal, setSshAccess] = useState(false)
+
+  const handleToggleSshAccess = (e) => {
+    e.preventDefault()
+    setSshAccess(e.target.checked)
+    toggleSshAccess()
+  }
+
+  const onSubmit = settings => {
+    updateSettings(settings)
+  }
+
+  // Leaving error logging until error states designed and implemented on form
+  console.log('!!! errors: ', errors)
+
+  return <div>
+    <div styleName='header'>
+      <span styleName='title'>HoloPort Settings</span>
+      <Button onClick={goToMenu} styleName='menu-button'>Menu</Button>
+    </div>
+    <div>
+      <label>Name</label>
+      <p>{settings.deviceName}</p>
+
+      <label>URL</label>
+      <p>{settings.hostUrl}</p>
+
+      <label>URL</label>
+      <p>{settings.networkId}</p>
+
+      <h2>Access Port Numbers</h2>
+
+      <form styleName='settings-form' onSubmit={handleSubmit(onSubmit)}>
+        <SettingsFormInput
+          label='Device Admin'
+          name='deviceAdminPort'
+          register={register} />
+
+        <SettingsFormInput
+          label='HC Admin'
+          name='hcAdminPort'
+          register={register} />
+
+        <SettingsFormInput
+          label='HC Network'
+          name='hcNetworkPort'
+          register={register} />
+
+        <SettingsFormInput
+          label='Hosting'
+          name='hostingPort'
+          register={register} />
+
+        <Button type='submit' name='update-settings' value='Submit'>Update</Button>
```

Comment:
This is our Button react component, not a native HTML button, which currently doesn't take `type`, `submit` or `value` props. I'm modifying this component to take type and value in my branch.